WebApi/Controllers/AuthorsController.cs

```diff
@@ -0,0 +1,134 @@
+    public class AuthorsController : ControllerBase
+    {
+        /// <summary>
+        /// ILibrary interface link
+        /// </summary>
+        private readonly IAuthor library;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorsController"/> class
+        /// </summary>
+        /// <param name="library">IBook interface link</param>
+        public AuthorsController(ILibrary library)
+        {
+            this.library = library;
+        }
+
+        /// <summary>
+        /// Adding author
+        /// </summary>
+        /// <param name="author">Author to add</param>
+        /// <returns>Result of adding</returns>
+        [HttpPost]
+        public IActionResult AddAuthor(Author author)
+        {
+            if (ModelState.IsValid)
+            {
+                this.library.AddAuthor(author);
+                return this.Created("authors", author);
+            }
+
+            return this.BadRequest("Invalid author format");
+        }
+
+        /// <summary>
+        /// Getting all authors
+        /// </summary>
+        /// <returns>Authors list</returns>
+        [HttpGet]
+        public IActionResult GetAllAuthors()
+        {
+            return this.Ok(this.library.GetAllAuthors().ToList());
+        }
+
+        /// <summary>
+        /// Getting author by id
+        /// </summary>
+        /// <param name="id">Id of author for get</param>
+        /// <returns>Author with such id or null if not exist</returns>
+        [HttpGet("{id}")]
+        public IActionResult GetAuthorById(int id)
+        {
+            Author authorToGet = this.library.GetAuthorById(id);
+            if (authorToGet == null)
+            {
+                return this.NotFound("No author with this id!");
+            }
+
+            return this.Ok(authorToGet);
+        }
+
+        /// <summary>
+        /// Getting book list of author
+        /// </summary>
+        /// <param name="id">Id of author</param>
+        /// <returns>List of books or NotFound</returns>
+        [HttpGet("{id}/books")]
+        public IActionResult GetBooksOfAuthor(int id)
+        {
+            Author authorToGet = this.library.GetAuthorById(id);
+            if (authorToGet == null)
+            {
+                return this.NotFound("No author with this id!");
+            }
+
+            return this.Ok(this.library.GetAuthorBooks(id).ToList());
+        }
+
+        /// <summary>
+        /// Updating of author
+        /// </summary>
+        /// <param name="id">Id of author for upadte</param>
+        /// <param name="author">New info</param>
+        /// <returns>Status of update</returns>
+        [HttpPut("{id}")]
+        public IActionResult UpdateAuthor(int id, [FromBody]Author author)
+        {
+            if (!ModelState.IsValid)
+            {
+                return this.BadRequest("Not a valid model!");
+            }
+
+            if (!this.library.UpdateAuthor(id, author))
```

Comment:
Mm try to avoid placing the most valuable method call according to the logic of the current method in the if statement. You are reducing the readability by this. 
It will be okay to have a bool result variable that you will use only for this if.